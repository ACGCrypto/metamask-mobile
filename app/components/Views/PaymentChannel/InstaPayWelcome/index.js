import React, { PureComponent } from 'react';
import { SafeAreaView, Image, Text, View, StyleSheet, Platform } from 'react-native';
import { colors, fontStyles } from '../../../../styles/common';
import PropTypes from 'prop-types';
import StyledButton from '../../../UI/StyledButton';
import DeviceSize from '../../../../util/DeviceSize';
import { strings } from '../../../../../locales/i18n';
import AsyncStorage from '@react-native-community/async-storage';

const styles = StyleSheet.create({
	container: {
		flex: 1,
		backgroundColor: colors.white,
		paddingBottom: DeviceSize.isIphoneX() ? 32 : 8
	},
	frame: {
		justifyContent: 'center',
		marginTop: DeviceSize.isSmallDevice() ? 5 : 80,
		marginBottom: 10,
		marginHorizontal: 35,
		alignSelf: 'center',
		width: DeviceSize.isSmallDevice() ? '80%' : '100%'
	},
	content: {
		flex: 1,
		marginHorizontal: 25,
		marginVertical: DeviceSize.isIphoneX() ? 10 : 0
	},
	text: {
		...fontStyles.normal,
		fontSize: 16,
		lineHeight: 24,
		color: colors.fontPrimary,
		textAlign: 'center'
	},
	text2: {
		marginTop: 30
	},
	title: {
		fontSize: 24,
		...fontStyles.bold,
		color: colors.fontPrimary,
		textAlign: 'center',
		marginBottom: 20
	},
	button: {
		flexDirection: 'row',
		alignSelf: 'center',
		width: '100%'
	},
	buttonWrapper: {
		margin: 30,
		marginBottom: 30
	},
	spacer: {
		marginTop: Platform.OS === 'ios' ? 24 : 14,
		height: 50
	}
});

const welcomeImage = require('../../../../images/payment-channel-welcome.png'); // eslint-disable-line import/no-commonjs

/**
 * View show a welcome screen for payment channels
 */
export default class InstaPayWelcome extends PureComponent {
	static propTypes = {
		/**
		 * The navigator object
		 */
		navigation: PropTypes.object
	};

	constructor(props) {
		super(props);
		this.state = { migrated: false };
	}

	componentDidMount() {
		this.checkMigrationStatus();
	}

	checkMigrationStatus = async () => {
		const instapayVersion = await AsyncStorage.getItem('@MetaMask:InstaPayVersion');
		if (!instapayVersion) {
			//this.setState({migrated: true});
		}
	};

	next = () => {
		this.props.navigation.navigate('InstaPayUpdate');
	};

	close = () => {
		this.props.navigation.popToTop();
		this.props.navigation.goBack(null);
	};

	render() {
		return (
			<SafeAreaView style={styles.container}>
				<View style={styles.content}>
					<View style={styles.spacer} />
					<Text style={styles.title}>{strings('payment_channel.welcome.title')}</Text>
					<Text style={styles.text}>{strings('payment_channel.welcome.desc_1')}</Text>
					<Text style={[styles.text, styles.text2]}>{strings('payment_channel.welcome.desc_2')}</Text>
					<Image source={welcomeImage} style={styles.frame} resizeMode={'contain'} />
				</View>
				<View style={styles.buttonWrapper}>
					{this.state.migrated ? (
						<StyledButton type={'normal'} onPress={this.close} containerStyle={styles.button}>
							{strings('payment_channel.welcome.close')}
						</StyledButton>
					) : (
						<StyledButton type={'normal'} onPress={this.next} containerStyle={styles.button}>
							{strings('payment_channel.welcome.next')}
						</StyledButton>
					)}
				</View>
			</SafeAreaView>
		);
	}
}
