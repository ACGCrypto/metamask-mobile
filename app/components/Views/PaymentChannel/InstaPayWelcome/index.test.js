import React from 'react';
import { shallow } from 'enzyme';
import InstaPayWelcome from './';
import configureMockStore from 'redux-mock-store';

const mockStore = configureMockStore();

describe('InstaPayWelcome', () => {
	it('should render correctly', () => {
		const initialState = {
			engine: {
				backgroundState: {}
			}
		};

		const wrapper = shallow(<InstaPayWelcome />, {
			context: { store: mockStore(initialState) }
		});
		expect(wrapper).toMatchSnapshot();
	});
});
