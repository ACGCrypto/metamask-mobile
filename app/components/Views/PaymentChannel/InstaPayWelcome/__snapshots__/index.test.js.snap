// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstaPayWelcome should render correctly 1`] = `
<SafeAreaView
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "paddingBottom": 32,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginHorizontal": 25,
        "marginVertical": 10,
      }
    }
  >
    <View
      style={
        Object {
          "height": 50,
          "marginTop": 24,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#000000",
          "fontFamily": "Roboto",
          "fontSize": 24,
          "fontWeight": "600",
          "marginBottom": 20,
          "textAlign": "center",
        }
      }
    >
      Welcome to InstaPay 2.0
    </Text>
    <Text
      style={
        Object {
          "color": "#000000",
          "fontFamily": "Roboto",
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
          "textAlign": "center",
        }
      }
    >
      Now deposit up to 100 DAI, with no minimum required. Simplified account management and improved user experience.
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Roboto",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 24,
            "textAlign": "center",
          },
          Object {
            "marginTop": 30,
          },
        ]
      }
    >
      Always non-custodial and built to the latest standards of state channel technology.
    </Text>
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../app/images/payment-channel-welcome.png",
        }
      }
      style={
        Object {
          "alignSelf": "center",
          "justifyContent": "center",
          "marginBottom": 10,
          "marginHorizontal": 35,
          "marginTop": 80,
          "width": "100%",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "margin": 30,
        "marginBottom": 30,
      }
    }
  >
    <StyledButton
      containerStyle={
        Object {
          "alignSelf": "center",
          "flexDirection": "row",
          "width": "100%",
        }
      }
      disabledContainerStyle={
        Object {
          "opacity": 0.6,
        }
      }
      onPress={[Function]}
      styleDisabled={
        Object {
          "opacity": 0.6,
        }
      }
      type="normal"
    >
      Next
    </StyledButton>
  </View>
</SafeAreaView>
`;
